backend/app/Http/Controllers/Api/V1/AccommodationController.php

```diff
@@ -102,27 +101,38 @@ public function store(Request $request) {
         });
     }
 
-    private function getValidationRules() {
-        return [
-            'type' => ['required', 'string', 'in:'.implode(',', AccommodationType::getConstants())],
-            'bed_count' => 'required|integer|min:1',
-            'bedroom_count' => 'required|integer|min:1',
-            'bathroom_count' => 'required|integer|min:1',
-            'minimum_days' => 'required|integer|min:1',
-            'maximum_days' => 'required|integer|min:1',
-            'amenities' => ['array', 'in:'.implode(',', Amenity::getConstants())],
-            'name' => 'required|string',
-            'description' => 'required',
-            'province' => 'required|string',
-            'city' => 'required|string',
-            'barangay' => 'required|string',
-            'street' => 'required|string',
-            'zip_code' => 'required|numeric',
-            'price' => 'required|string',
-            'maximum_guests' => 'required|integer|min:1',
-            'media' => 'required|min:1',
-            'media.*' => 'url',
-        ];
+    public function update(AccomodationUpdateRequest $request, $listingId) {
```

Comment:
try to breakdown into other functions and move some logic to the accommodation model.